libp2p/src/main/kotlin/io/libp2p/mux/yamux/YamuxHandler.kt

```diff
@@ -87,11 +91,23 @@ open class YamuxHandler(
                     writeAndFlushFrame(YamuxFrame(msg.id, YamuxType.WINDOW_UPDATE, YamuxFlag.ACK.asSet, 0))
                 }
 
+                YamuxFlag.ACK in msg.flags -> {
+                    if (outbound) {
+                        acknowledged.set(true)
+                    }
```

Comment:
nit: May be extract to function `acknowledgeOutboundStreamIfNeeded()` for uniformity?